addons/sale/report/sale_report.py

```diff
@@ -62,42 +62,42 @@ def _select_sale(self, fields=None):
-            CASE WHEN l.product_id IS NOT NULL THEN sum(l.qty_delivered / u.factor * u2.factor) ELSE 0 END as qty_delivered,
-            CASE WHEN l.product_id IS NOT NULL THEN SUM((l.product_uom_qty - l.qty_delivered) / u.factor * u2.factor) ELSE 0 END as qty_to_deliver,
-            CASE WHEN l.product_id IS NOT NULL THEN sum(l.qty_invoiced / u.factor * u2.factor) ELSE 0 END as qty_invoiced,
-            CASE WHEN l.product_id IS NOT NULL THEN sum(l.qty_to_invoice / u.factor * u2.factor) ELSE 0 END as qty_to_invoice,
-            CASE WHEN l.product_id IS NOT NULL THEN sum(l.price_total / CASE COALESCE(s.currency_rate, 0) WHEN 0 THEN 1.0 ELSE s.currency_rate END) ELSE 0 END as price_total,
-            CASE WHEN l.product_id IS NOT NULL THEN sum(l.price_subtotal / CASE COALESCE(s.currency_rate, 0) WHEN 0 THEN 1.0 ELSE s.currency_rate END) ELSE 0 END as price_subtotal,
-            CASE WHEN l.product_id IS NOT NULL THEN sum(l.untaxed_amount_to_invoice / CASE COALESCE(s.currency_rate, 0) WHEN 0 THEN 1.0 ELSE s.currency_rate END) ELSE 0 END as untaxed_amount_to_invoice,
-            CASE WHEN l.product_id IS NOT NULL THEN sum(l.untaxed_amount_invoiced / CASE COALESCE(s.currency_rate, 0) WHEN 0 THEN 1.0 ELSE s.currency_rate END) ELSE 0 END as untaxed_amount_invoiced,
-            count(*) as nbr,
-            s.name as name,
-            s.date_order as date,
-            s.state as state,
-            s.partner_id as partner_id,
-            s.user_id as user_id,
-            s.company_id as company_id,
-            s.campaign_id as campaign_id,
-            s.medium_id as medium_id,
-            s.source_id as source_id,
-            extract(epoch from avg(date_trunc('day',s.date_order)-date_trunc('day',s.create_date)))/(24*60*60)::decimal(16,2) as delay,
-            t.categ_id as categ_id,
-            s.pricelist_id as pricelist_id,
-            s.analytic_account_id as analytic_account_id,
-            s.team_id as team_id,
+            COALESCE(min(l.id), -s.id) AS id,
+            l.product_id AS product_id,
+            t.uom_id AS product_uom,
+            CASE WHEN l.product_id IS NOT NULL THEN SUM(l.product_uom_qty / u.factor * u2.factor) ELSE 0 END AS product_uom_qty,
+            CASE WHEN l.product_id IS NOT NULL THEN SUM(l.qty_delivered / u.factor * u2.factor) ELSE 0 END AS qty_delivered,
+            CASE WHEN l.product_id IS NOT NULL THEN SUM((l.product_uom_qty - l.qty_delivered) / u.factor * u2.factor) ELSE 0 END AS qty_to_deliver,
+            CASE WHEN l.product_id IS NOT NULL THEN SUM(l.qty_invoiced / u.factor * u2.factor) ELSE 0 END AS qty_invoiced,
+            CASE WHEN l.product_id IS NOT NULL THEN SUM(l.qty_to_invoice / u.factor * u2.factor) ELSE 0 END AS qty_to_invoice,
+            CASE WHEN l.product_id IS NOT NULL THEN SUM(l.price_total * CASE COALESCE(currency_table.rate, 0) WHEN 0 THEN 1.0 ELSE currency_table.rate END) ELSE 0 END AS price_total,
```

Comment:
I will change it. Your argument now and the one discussed and the office seems more correct now to me. It was that way before also.

Reply:
I think there is a slight misunderstanding here.

These are two different cases that can combine:
1. To make a report, you convert every SO to your company currency => currency rate of the SO (existing flow, the one I don't want changed)
2. You want your existing report in another currency than its company's one => you need to use the currency rate of the day to "translate" it and aggregate your values to your second/third/... companies

But if you ask for the report in the company's currency, you still shouldn't use the rate of the day, IMHO.

But it's better to ask confirmation to the product owner(s) for this, to be sure of what exactly is expected.

Reply:
You have to "first" compute the company report in its currency using 
- US company: if there are sales made in another currency than the $ for this company, it must be converted using the 'sale_order.currency_rate'
- EU company: if there are sales made in another currency than the € for this company, it must be converted using the 'sale_order.currency_rate'

Now, if you want to display a combined report for those two company in $ (or any other currency) you will have to convert all amounts into this currency based on the 'currency_table.rate'

Reply:
Okay, it might seem too simple but the solution is to multiply the two rates:
1- for each SO multiply it by the SO currency rate to compute the amount of the SO on the currency of the company that created the SO, this way each SO has its amount computed correctly in the SO company currency
2- in the case of multi companies with different currencies then multiply the amount found in (1) by the current conversion rate to get the amount in the currency of the company selected currently on the reporting tab.
Thus to correctly compute the amount:
`so.amount * so.currency_rate * company_selected.conversion_rate`
The first multiplication set the amount to the SO company currency and the second set it to the currency of the company currently selected on the report.